back_end/database/update_db_helpers.py

```diff
@@ -0,0 +1,196 @@
+"""
+Contains helper methods used in update_db_*.py scripts to update term and course data in the database.
+"""
+
+import json
+from database import Database
+from mobileapp import MobileApp
+from registrar import RegistrarAPI
+import logging
+
+logging.basicConfig(format='%(asctime)s - %(levelname)s - %(name)s: %(message)s', datefmt='%m/%d/%Y %H:%M:%S', level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+# update course info for single term
+# collections updated: courses, instructors
+def update_courses_for_one_term(term: str) -> None:
+    db = Database()
+
+    # get course data from mobileapp api
+    try:
+        logger.info(f"getting course data for term {term} from mobileapp")
+        all_courses = MobileApp().get_all_courses(term)
+    except:
+        logger.error(f"unable to get course data for term {term} from mobileapp")
+        return
+
+    # NOTE: it's possible that courses can be deleted, instructors
+    # removed from a course. only do two clearing operations below 
+    # if confident that update for all courses will not fail.
+    # db.clear_courses_for_one_term(term)
+    # db.clear_courses_for_instructor_for_one_term(term)
```

Comment:
@howard-yen thoughts on lines 30-31 above? i found out that courses can be removed from MobileApp. however, Princeton Courses never handled this -- i.e. there are courses stored in their DB that have been removed from MobileApp (these courses are usually special, e.g. "Senior Thesis", "Junior IW", "Departmental Exam".

@howard-yen Should we handle removed courses? My main concern is that if we clear all courses (for one term) before re-inserting these courses (with the updated data), if something goes wrong mid-way in the loop below, then there will be missing course documents in the DB. If we don't clear courses before inserting, at least the old course data will still be in the DB.

A second approach would be to store the set of courseids from MobileApp, then once the loop finishes, delete courses that aren't in this stored set. This operation might be somewhat slow, but is probably safer. Let me know if we should do either approach.

Similar idea for clearing then updating the instructors collection.

Reply:
what does it mean for a course to be removed? is it originally listed on the registrar website and then later taken off?

Reply:
hmm, i'm starting to rethink my original comment. i will ask Jonathan Lebouef whether courses can ever be deleted from MobileApp and whether instructors can ever be removed from a course. for now, we'll assume that removing courses/instructors can't happen, but i'll see what Jonathan says.

FYI, here's an example of a course in Princeton Course's DB that is not in MobileApp, but is in Registrar's API: https://www.princetoncourses.com/course/1194002140?search=Senior%20Departmental%20Exam&semester=1192&sort=commonName. i think we should ignore cases like this because MobileApp seems like the source of truth.